proto/transport.proto

```diff
@@ -0,0 +1,224 @@
+      OVERSPEED_FAULT = 4;
+      POWER_SUPPLY_FAULT = 5;
+      SAFETY_DEVICE_FAULT = 6;
+      CONTROLLER_SUPPLY_FAULT = 7;
+      DRIVE_TEMPERATURE_EXCEEDED = 8;
+      // A comb plate safety switch is activated.
+      COMB_PLATE_FAULT = 9;
+      GENERAL_FAULT = 10;
+      DOOR_FAULT = 11;
+      LEVELLING_FAULT = 12;
+      SAFETY_CIRCUIT_BREAK_FAULT = 13;
+      FAIL_TO_START = 14;
+      ALARM_BUTTON = 15;
+      DOOR_NOT_CLOSING = 16;
+    }
+
+    // Location describes a place that the transport can stop at.
+    message Location {
+      // The id of the location.
+      string id = 1;
+      // The name of the location. e.g. 'Lobby', 'Floor 1', 'Car Park'.
+      string name = 2;
+      // Optional. Description of the location.
+      string description = 3;
+      // Optional. The floor number of the location.
+      int32 floor = 4;
+      // Optional. The attributes of the location.
+      repeated string attributes = 5;
+    }
+
+    // Journey describes the journey of the transport or payload.
+    // It will have a source and one or more destinations.
+    message Journey {
+      // The starting point of the journey.
+      Location start = 1;
+      // The destinations of the journey.
+      repeated Location destinations = 2;
+      // Optional. The reason for the journey.
+      string reason = 3;
+    }
+
+    // The current location of the transport.
+    Location current_location = 1;
+    // Optional. The next destination(s) the transport will stop at.
+    repeated Location next_destinations = 2;
+    // Optional. The direction the transport is moving.
+    Direction moving_direction = 3;
+    // Optional. The total amount of load being carried.
+    // The units this is measured in is implementation specific. See TransportSupport
+    float load = 4;
+    // Optional. The status of each the door(s) of the transport. For multiple doors, use TransportSupport to identify multiple doors.
+    repeated DoorStatus door_status = 5;
+    // Optional. The mode the transport is in.
+    Mode mode = 6;
+    // Optional. A list of active fault signals being reported.
+    repeated Fault faults = 7;
+    // Optional. True if the passenger alarm has been activated.
+    bool passenger_alarm = 8;
+    // Optional. The speed of the transport.
+    float speed = 9;
+    // Optional. The destinations that are supported by the transport.
+    repeated Location supported_destinations = 10;
+    // Optional. True if the transport is currently active. Note this is different to mode,
+    // as the transport can be in an active or idle state in different modes.
+    bool active = 11;
+
+    message Payload {
+        // The id of the payload.
+        string id = 1;
+        // The description of the payload.
+        string description = 2;
+        // The journey the payload is intending to take.
+        Journey intended_journey = 3;
+        // The actual journey the payload has taken.
+        Journey actual_journey = 4;
```

Comment:
As there's no way to get completed payloads, I don't think this will ever include the actual journey, only the actual journey so far.